### DISCUSSION TITLE
How to save many hasOne relations in Controller.store/update

### DISCUSSION THREAD

**Discussion by ekimkael** [adonisjs/core#4438] [Help]
Hello everyone,
I have a **Course** model who has `fulltime`. 

```ts
@hasOne(() => CourseOption)
public fulltime: HasOne<typeof CourseOption>
```

in `fulltime` I have 2 hasOne relations.

```ts
@hasOne(() => CourseFee)
public locals: HasOne<typeof CourseFee>

@hasOne(() => CourseFee)
public foreigners: HasOne<typeof CourseFee>
```
in my **controller** I want to save `locals` and `foreigners` so I'm doing this:

```ts
const course = await Course.query().select('id', 'name')
.preload('fulltime')
.where('slug', params.id)
.firstOrFail()

const payload = await request.validate(CourseOptionValidator)

await course.fulltime.related('locals').updateOrCreate({}, payload.locals)
await course.fulltime.related('foreigners').updateOrCreate({}, payload.foreigners)
```
but it saves locals for the 2 properties (fulltime.locals & fulltime.foreigners). How can I solve it?

**Comment by ekimkael** [comment]
any help please? @RomainLanz @thetutlage 🥲

**Comment by holoyan** [comment]
tbh, this doen't make sence for me,

```typescript
@hasOne(() => CourseFee)
public locals: HasOne<typeof CourseFee>

@hasOne(() => CourseFee)
public foreigners: HasOne<typeof CourseFee>

```
 you don't have to define relationships for each column, instead you should have for model, like this


```typescript
@hasOne(() => CourseFee)
public courseFee: HasOne<typeof CourseFee>

```

and then you can create relation
```typescript
await course.fulltime.related('courseFee').create(payload)

```

see [hasOne docs](https://lucid.adonisjs.com/docs/relationships#hasone)

also check [updateOrCreate](https://lucid.adonisjs.com/docs/crud-operations#updateorcreate) docs
